### ISSUE TITLE
Access hooks or runner from loss

### ISSUE THREAD

**Issue by ggggg111** [kennymckormick/pyskl#191]
Hello!

I would like to ask how to access the hook data from the loss function, as I need the data to perform in my loss function. I have investigated and I haven't found a way. What is the best workaround?

Beforehand, much appreciated for your help!

Here is the code:

```python
@HOOKS.register_module()
class OutputOuternHook(Hook):
    def __init__(self, outputs, as_tensor):
        self.outputs = outputs
        self.as_tensor = as_tensor

    def before_train_iter(self, runner):
	# OutputHook is defined in pyskl/core/hooks.py by default
        self.output_hook = OutputHook(runner.model, self.outputs, self.as_tensor)

    def after_train_iter(self, runner):
        self.output_hook.remove()

@LOSSES.register_module()
class CustomLoss(nn.Module):
    def __init__(self, test_data):
        super().__init__()

        self.test_data = mmcv.load(test_data)

    def forward(self, cls_score, label, **kwargs):
        for key, value in kwargs.items():
            print(f"{key}: {value}")

	# How to access the values of the output hooks? Can I access the hooks or runner from here?
        
	loss = ...

	return loss
```
